src/fabric/src/fabric_view_all_docs.erl

```diff
@@ -286,6 +286,20 @@ all_docs_concurrency() ->
             10
     end.
 
+validate_if_partition(Row, Acc) ->
+    QueryStr = Acc#vacc.req#httpd.qs,
+    case QueryStr of
+        undefined -> Row;
+        [{"partition", Id}] -> is_doc_in_partition(Row, Id)
+    end.
+is_doc_in_partition(Row, Id) ->
+    BinRowId = element(2, Row),
```

Comment:
the more idiomatic way of doing this in Erlang would be:

```
 {_, BinRowId, _, _, _, _} = Row,
```

but I’m not 100% that that’s actually better. I think maybe on revising the code it is nicer to know the explicit row structure rather than having to know the id is in location 2.

Reply:
Neither of those are good approaches. `Row#view_row.id` extracts the required field by name.

Reply:
thanks, that’s what I was looking for

Reply:
This is so much better, thank you!

Reply:
I realize now that I wasn't totally clear with my variable name as I am trying to get the key, not the id.  So I will update as follows:

```erlang
BinRowKey = Row#view_row.key,
```